src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/markduplicates/MarkDuplicatesSparkUtils.java

```diff
@@ -406,10 +406,10 @@ private static int countOpticalDuplicates(OpticalDuplicateFinder finder, List<Pa
                     metrics.updateMetrics(read);
                     // NOTE: we use the SAMRecord transientAttribute field here specifically to prevent the already
                     // serialized read from being parsed again here for performance reasons.
-                    if (((SAMRecordToGATKReadAdapter) read).getTransientAttribute(OPTICAL_DUPLICATE_TOTAL_ATTRIBUTE_NAME)!=null) {
+                    if (read.getTransientAttribute(OPTICAL_DUPLICATE_TOTAL_ATTRIBUTE_NAME)!=null) {
```

Comment:
There is a corresponding `setTransientAttribute` call in `MarkDuplicatesSpark` that has a redundant `SAMRecordToGATKReadAdapter` cast that can now be removed.

Reply:
Thank you, I missed that.